src/theory/model_manager_distributed.cpp

```diff
@@ -81,11 +81,17 @@ bool ModelManagerDistributed::prepareModel()
       continue;
     }
     Theory* t = d_te.theoryOf(theoryId);
+    if (t->getTheoryState() == nullptr)
```

Comment:
I'd prefer a different way of enforcing this policy.  I'd actually be more happy with just special casing for BOOL and BUILTIN here.

Reply:
Done.